src/main/java/org/scijava/io/DataHandle.java

```diff
@@ -74,43 +82,86 @@
 	void setLength(long length) throws IOException;
 
 	/**
+	 * Verifies that the handle has sufficient bytes available to read, returning
+	 * the actual number of bytes which will be possible to read, which might
+	 * be less than the requested value.
+	 * 
+	 * @param count Number of bytes to read.
+	 * @return The actual number of bytes available to be read.
+	 * @throws IOException If something goes wrong with the check.
+	 */
+	default long available(final long count) throws IOException {
```

Comment:
I guess for writing, the length can of course change. My comments above apply to reading. Hmm, thoughts?